datalad_container/containers_run.py

```diff
@@ -133,39 +144,54 @@ def __call__(cmd, container_name=None, dataset=None,
             # just prepend and pray
             cmd = container['path'] + ' ' + cmd
 
+        assume_ready = assume_ready or []
         extra_inputs = []
-        for extra_input in ensure_iter(container.get("extra-input",[]), set):
-            try:
-                xi_kwargs = dict(
-                    img_dspath=image_dspath,
-                    img_dirpath=op.dirname(image_path) or ".",
-                )
-                extra_inputs.append(extra_input.format(**xi_kwargs))
-            except KeyError as exc:
-                yield get_status_dict(
-                    'run',
-                    ds=ds,
-                    status='error',
-                    message=(
-                        'Unrecognized extra_input placeholder: %s. '
-                        'See containers-add for information on known ones: %s',
-                        exc,
-                        ", ".join(xi_kwargs)))
-                return
+        if not "extra-inputs" in assume_ready:
+            for extra_input in ensure_iter(container.get("extra-input",[]), set):
+                try:
+                    xi_kwargs = dict(
+                        img_dspath=image_dspath,
+                        img_dirpath=op.dirname(image_path) or ".",
+                    )
+                    extra_inputs.append(extra_input.format(**xi_kwargs))
+                except KeyError as exc:
+                    yield get_status_dict(
+                        'run',
+                        ds=ds,
+                        status='error',
+                        message=(
+                            'Unrecognized extra_input placeholder: %s. '
+                            'See containers-add for information on known ones: %s',
+                            exc,
+                            ", ".join(xi_kwargs)))
+                    return
+        assume_ready = [ar for ar in assume_ready if ar != 'extra-inputs']
+
+        if not "image" in assume_ready:
+            extra_inputs.append(image_path)
+        assume_ready = [ar for ar in assume_ready if ar != 'image']
 
         lgr.debug("extra_inputs = %r", extra_inputs)
 
+        if len(assume_ready) == 0:
+            assume_ready = None
+        elif len(assume_ready) == 1:
+            assume_ready = assume_ready[0]
+        elif  "inputs" in assume_ready and "outputs" in assume_ready:
+            assume_ready = "both"
```

Comment:
little of generic coding + proofing:

```suggestion
        if not assume_ready:
            assume_ready = None
        elif len(assume_ready) == 1:
            assume_ready = assume_ready[0]
        elif  "inputs" in assume_ready and "outputs" in assume_ready:
            assume_ready = "both"
        else:
            raise ValueError(f"Ended up with assume_ready={assume_ready!r} which must have not happened")
```